### ISSUE TITLE
[intelliJ][Report to Microsoft][SQL Server big data]Get error when submit and remotely run with 52.179.143.122 cluster.

### ISSUE THREAD

**Issue by v-jiche** [microsoft/azure-tools-for-java#3534]
Build:
develop 1247

Repro Steps:
1. Choose application type as aris
2. Right click LogQuery to submit
![image](https://user-images.githubusercontent.com/35950097/64751523-f076e580-d54e-11e9-8684-564e5f715186.png)
Result:
![image](https://user-images.githubusercontent.com/35950097/64751472-c32a3780-d54e-11e9-94ad-9c378357c94c.png)

```
rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: com.microsoft.azure.hdinsight.sdk.common.livy.interactive.SparkSession.class
	at rx.exceptions.OnErrorThrowable.addValueAsLastCause(OnErrorThrowable.java:118)
	at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:73)
	at rx.observers.Subscribers$5.onNext(Subscribers.java:235)
	at rx.internal.operators.OperatorDoAfterTerminate$1.onNext(OperatorDoAfterTerminate.java:50)
	at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:77)
	at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:77)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.emitScalar(OperatorMerge.java:395)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.tryEmit(OperatorMerge.java:355)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onNext(OperatorMerge.java:846)
	at rx.observers.Subscribers$5.onNext(Subscribers.java:235)
	at rx.internal.operators.OperatorDoAfterTerminate$1.onNext(OperatorDoAfterTerminate.java:50)
	at rx.internal.util.ScalarSynchronousObservable$WeakSingleProducer.request(ScalarSynchronousObservable.java:276)
	at rx.Subscriber.setProducer(Subscriber.java:211)
	at rx.Subscriber.setProducer(Subscriber.java:205)
	at rx.Subscriber.setProducer(Subscriber.java:205)
	at rx.internal.util.ScalarSynchronousObservable$JustOnSubscribe.call(ScalarSynchronousObservable.java:138)
	at rx.internal.util.ScalarSynchronousObservable$JustOnSubscribe.call(ScalarSynchronousObservable.java:129)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeUsing.call(OnSubscribeUsing.java:94)
	at rx.internal.operators.OnSubscribeUsing.call(OnSubscribeUsing.java:32)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:248)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:148)
	at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:77)
	at rx.internal.producers.SingleDelayedProducer.emit(SingleDelayedProducer.java:102)
	at rx.internal.producers.SingleDelayedProducer.setValue(SingleDelayedProducer.java:85)
	at rx.internal.operators.OnSubscribeFromCallable.call(OnSubscribeFromCallable.java:48)
	at rx.internal.operators.OnSubscribeFromCallable.call(OnSubscribeFromCallable.java:33)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeUsing.call(OnSubscribeUsing.java:94)
	at rx.internal.operators.OnSubscribeUsing.call(OnSubscribeUsing.java:32)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.Observable.subscribe(Observable.java:10423)
	at rx.Observable.subscribe(Observable.java:10390)
	at rx.observables.BlockingObservable.blockForSingle(BlockingObservable.java:443)
	at rx.observables.BlockingObservable.single(BlockingObservable.java:340)
	at com.microsoft.azure.hdinsight.spark.jobs.JobUtils.uploadFileToHDFS(JobUtils.java:608)
	at com.microsoft.azure.hdinsight.spark.jobs.JobUtils.uploadFileToCluster(JobUtils.java:622)
	at com.microsoft.azure.hdinsight.spark.jobs.JobUtils.lambda$deployArtifact$11(JobUtils.java:679)
	at rx.Single.subscribe(Single.java:1979)
	at rx.internal.operators.SingleOnSubscribeMap.call(SingleOnSubscribeMap.java:45)
	at rx.internal.operators.SingleOnSubscribeMap.call(SingleOnSubscribeMap.java:30)
	at rx.internal.operators.SingleToObservable.call(SingleToObservable.java:39)
	at rx.internal.operators.SingleToObservable.call(SingleToObservable.java:27)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:41)
	at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:30)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OperatorSubscribeOn$SubscribeOnSubscriber.call(OperatorSubscribeOn.java:100)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	at rx.internal.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:107)
	at com.microsoft.intellij.rxjava.IdeaSchedulers$1.run(IdeaSchedulers.java:51)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:894)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:169)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:156)
	at com.intellij.openapi.progress.impl.CoreProgressManager$4.run(CoreProgressManager.java:408)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:294)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by v-jiche** [comment]
Use 52.179.143.122 to run interactive console
Result:
![image](https://user-images.githubusercontent.com/35950097/64754806-6719e000-d55b-11e9-8dba-bf94f859ab43.png)

**Comment by t-rufang** [comment]
The exeception happens when run livy interactive console is by design. However, the exception happens when submit job to aris cluster is not by design. We should catch the exception and expose some meaningful error message just as the error message when run livy interactive console.

This issue is fixed in #3638

**Comment by v-jiche** [comment]
Can not link aris cluster successfully.
![image](https://user-images.githubusercontent.com/35950097/72045107-4d618d00-32f0-11ea-9725-ebc45f30b9cd.png)

**Comment by sunilm-css** [comment]
I am getting this Timed out. 
What is server? here

Cluster controller endpoint is available at 52.186.120.123:30080.
SQL Server Master Instance Front-End                    52.188.123.124,31433
Gateway to access HDFS files, Spark                     https://52.188.123.125:30443
...
I tried SQL server 120.123 and also SqlServerMaster 123.124. Neither of them worked.

This is blocking.

**Comment by t-rufang** [comment]
@sunilm-css Sorry for late reply, I didn't notice the comment here. I suppose you can try server `52.188.123.125` with knox port `30443` considering we are using gateway endpoint to link cluster.
